executable-spec/src/Cardano/Ledger/Spec/STS/Common/Crypto.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Cardano.Ledger.Spec.STS.Common.Crypto where
+
+import           Data.Either (isRight)
+import qualified Codec.CBOR.Write as CBOR.Write
+import qualified Crypto.Hash as Crypto
+import qualified Data.ByteString.Builder as Builder
+import qualified Data.ByteString.Lazy as BSL
+
+import           Cardano.Binary (ToCBOR, toCBOR)
+import           Cardano.Crypto.DSIGN.Class (SignedDSIGN)
+import qualified Cardano.Crypto.DSIGN.Class as Crypto.DSIGN
+import           Cardano.Crypto.DSIGN.Mock (MockDSIGN)
+import           Cardano.Crypto.DSIGN.Mock (SignKeyDSIGN (SignKeyMockDSIGN),
+                     VerKeyDSIGN (VerKeyMockDSIGN))
+import qualified Cardano.Crypto.DSIGN.Mock as Crypto.Mock
+
+import           Cardano.Ledger.Spec.Classes.Hashable (HasHash, Hash, Hashable,
+                     hash)
+import           Cardano.Ledger.Spec.Classes.HasSigningScheme (HasSigningScheme,
+                     SKey, Signable, Signature, VKey, sign, verify)
+import qualified Crypto.Hash as Crypto
+
+--------------------------------------------------------------------------------
+-- Hashing, signing, and verification algorithms to be used in the benchmarks
+--------------------------------------------------------------------------------
+
+data BenchCrypto
+
+-- deriving instance ( ToCBOR (Crypto.Digest Crypto.Blake2b_224)
+--                   ) => ToCBOR BenchCrypto
+
+instance Hashable BenchCrypto where
+
+  newtype Hash BenchCrypto a = BenchHash (Crypto.Digest Crypto.Blake2b_224)
```

Comment:
So I would simply change this to `BenchHash ShortBytestring` and change the `hash` implementation to translate the hash to a short bytestring.